Zombies!

ErhardSteinhauer opened the pull request:
## About the PR
- Adds a new expedition enemy type — zombies. Unlike real zombies, these ones cannot infect players. Limited random visuals for the corpses.
  - Zombies deal primarily slash type of damage and inject 1u of zombie bite toxin with each attack: after approximately 3–4 bites, the affected character suffers a 15% movement slowdown. After about 8 bites, the toxin begins to metabolize and inflict additional poison damage. Enemies have weakness to heat damage type. When butchered drop rotten meat and random valuable item.
  - Zombies have powerful health regeneration and will rise from the dead after some time. Butchering them prevents resurrection.
  - Six "special" zombies: slightly radioactive zombie (wears torn radiation protection suit), clown zombie (throws pies), flaming zombie (sets victims on fire in melee), cold zombie (throws freezing globes), cargo zombie (on melee hit pulls target in, explodes on death), fast zombie (faster than standard zombies but still slower than an unencumbered player character).
  - Dungeon boss: zombie trapped in a medical mecha, spawns body bag pile every 4 minutes.
  - Body bag pile: spawn 1 to 3 zombies every 5 minutes.
- Adds zombie themed VGRoid.
- Adds new armor suit: mercenary armored bio suit. Available from bounty vend.

## Why / Balance
Halloween? But also I think it's fine if the faction will stay after.

## Technical details
yml

## How to test
1. Spawn in, use entity spawn menu, search for `zombie`.
2. Spam expeditions until zombies come up.
3. Use `addgamerule BluespaceDungeonCaveZombie`

## Media
<img width="711" height="459" alt="image" src="https://github.com/user-attachments/assets/bade32be-7691-4061-8c3b-b847ac08555f" />

<img width="291" height="197" alt="image" src="https://github.com/user-attachments/assets/6b6dd6f7-4f8f-4e67-8690-eead4b60bf46" />

<img width="277" height="237" alt="image" src="https://github.com/user-attachments/assets/7c3d684c-f949-4e51-bedd-47faeb4bc8da" />

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [x] I have read [CONTRIBUTING.md](https://github.com/new-frontiers-14/frontier-station-14/blob/master/CONTRIBUTING.md) and and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.
- [x] I have reviewed the [Ship Submission Guidelines](https://frontierstation.wiki.gg/wiki/Ship_Submission_Guidelines) if relevant.
- [x] I confirm that the content in this PR is my own work, and/or is properly attributed to the original author(s).
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them. -->

**Changelog**
:cl:
- add: Adds a new expedition enemy type - zombies. Unlike real zombies, these ones cannot infect players or NPCs.
- add: Adds new armor suit: mercenary armored bio suit. Available from bounty vend.

JojyAsmir commented:
Hell yeah! I was actually gonna request a prototype for a random urist zombie spawned, but this is even better!

ErhardSteinhauer commented:
Added a VGRoid with zombies too. I think this PR is done and ready for review.

dustylens commented:
He did it, the madlad did it.

ErhardSteinhauer commented:
Added three more special zombies:
- flaming zombie (sets victims on fire on melee hits);
- cold zombie (throws freezing projectiles);
- cargo zombie (on melee hit pulls target in, explodes on death).

Tested whether consuming or injecting zombie blood turns a character into a zombie: it does not.

ErhardSteinhauer commented:
Actually had time to do some tweaks now, but no time for testing.
- Adjusted damage: removed toxin injection; now deals direct poison damage, mitigated by the new biosuit.
- Slightly reduced regeneration from brute damage group; significantly reduced regeneration from burn damage group (now regenerates at half the brute rate).
- Marginally increased loot drops.
- Reduced pile spawn rates: 3 min → 5 min; decreased the number of zombies spawned per pile.
- Decreased boss pile spawn frequency: 3 min → 4 min.

dustylens commented:
I ran a few VG zombie roids this evening to test out how things feel. As I spend pretty much 0 time doing things with guns in this game I'm probably as decent a person to give feedback as any. I had to take a moment to remember how to reload my Nova... Light? I think that's what it's called.

Posted up in a Dragoon. Sped on over. Ventured in with some laser pistols and basic rifle in hand.

Some impressions:

- Assuming you're not running with a light source we're fairly atmospheric. I feel like there might be room in the future for a little additional zombie soundage.
- Murdering zombies is pretty good. Gotta put 'em down for real.
- A bunch of zombies ooking at you down a hallway is not bad.
- Zombies popping up behind you is unwelcome and a good time.
- Zombie toxin buildup is a pretty hard counter to the glaive meta. Nice way to mix things up.

- Healing toxin damage is going to be challenging on site, but is very approachable with a medical bed back on the shuttle. Go get something for poison from a doc.

Challenge concerns.
- I feel that the zomies at present are pretty managable. I have absolutely no real combat time outside of punching rock monsters and I felt like I was able to clear out the entire VG roid pretty easily.
- Ran into a clown. I feel like the chances of this one popping up could be lowered a bit. A bit -_-. Didn't get a chance to see the others.
- I think two things that might make things feel more challenging, for me at any rate are
1) A slightly less common faster zombie. To break up the "simply walk away" strat and mix up the need to simply antagonize a bunch until they walk towards you. 
2) Perhaps some kind of grappling or slowdown inheret to the toxin. Extracting myself from zombies was not particularly challenging.
3) A little more health when they revive would be amazing, if we could swing it. I had an honestly phenominal moment where I lowskilled my way into getting chomped on and was trying to get my ass back to life in a room I cornered myself in:

![ss+(2025-11-11+at+07 32 11)](https://github.com/user-attachments/assets/e938bf6e-88b6-4f7c-b5eb-657461de088f)

But as I had killed all of them before they went down in one bullet. I think it would be much better if they came back with a lot more juice in the tank.

All in all I think that these zombies feel really good. I think that the challenge dial needs to be turned up considerably. With staggered faster zombies being an option that comes to mind and increased health from resurrected zombies.

But all in all I think that this PR looks really phenomenal.

ErhardSteinhauer commented:
>  increased health from resurrected zombies.

This will not happen without C#

marlyn-x86 commented:
There's another runtime that's been vexing me, but it seems to have to do with maploading the portable flashers in general (I get it with other palettes, too), so it's not this PR's responsibility

marlyn-x86 reviewed (approved):
Works good under playtest!
